astropy_timeseries/downsample.py

```diff
@@ -12,11 +12,16 @@
 
 def reduceat(array, indices, function):
     """
-    Manual reduceat functionality for cases where Numpy functions don't have a reduceat
+    Manual reduceat functionality for cases where Numpy functions don't have a reduceat.
+    It will check if the input function has a reduceat and call that if it does.
     """
-    result = [function(array[indices[i]:indices[i+1]]) for i in range(len(indices) - 1)]
-    result.append(function(array[indices[-1]:]))
-    return np.array(result)
+    # Taken some inspiration from TableGroups.aggregate.
```

Comment:
So in that function there is this logic and i'm wondering if we can use it in a simlar manner here:
```python
        sum_case = func is np.sum
        mean_case = func is np.mean
        try:
            if not masked and (reduceat or sum_case or mean_case):
                if mean_case:
                    vals = np.add.reduceat(par_col, i0s) / np.diff(self.indices)
                else:
                    if sum_case:
                        func = np.add
                    vals = func.reduceat(par_col, i0s)
            else:
                vals = np.array([func(par_col[i0: i1]) for i0, i1 in zip(i0s, i1s)])
```